lib/testsuite/repl.js

```diff
@@ -0,0 +1,143 @@
+const repl = require('repl');
+const {Logger} = require('../utils');
+const vm = require('vm');
+
+module.exports = class NightwatchRepl {
+  constructor(config) {
+    this._config = Object.assign({
+      eval: this._eval.bind(this),
+      useGlobal: false, // so that REPL does not create a separate context to pass to eval.
+      preview: true,
+      timeout: 5500  // timeout for assertions is 5000 ms.
+    }, config);
+  }
+
+  static introMessage() {
+    return `
+  DEBUG MODE on...
+
+  Type any Nightwatch command to execute in the browser in real-time.
+  Try, ${Logger.colors.cyan('browser.navigateTo(\'https://nightwatchjs.org\');')}
+
+  (To exit, press Ctrl+C twice or type .exit)
+  `;
+  }
+
+  startServer(context) {
+    vm.createContext(context);
+    this._context = context;
+
+    // Start REPLServer
+    this._replServer = repl.start(this._config);
+  }
+
+  _eval(cmd, _, filename, callback) {
+    // Evaluate for output previews.
+    if (/^try { .+ } catch {}$/.test(cmd)) {
+      // if `cmd` represents a method call
+      if (cmd.includes('(')) {
+        return;
+      }
+
+      return this._outputPreview(cmd, callback);
+    }
+
+    // A command is already running whose result is awaited.
+    if (this._resultAwaited) {
+      return;
+    }
+
+    // Make any modifications to the command before sending it over to vm.
+    if (cmd.includes('assert')) {
+      cmd = cmd.replaceAll('assert', 'verify');
+    }
+
+    try {
+      const result = vm.runInContext(cmd, this._context);
+      this._handleResult(result, callback);
+    } catch (err) {
+      const errRegex = /^(Unexpected end of input|Unexpected token)/;
+      if (err.name === 'SyntaxError' && errRegex.test(err.message)) {
+        return callback(new repl.Recoverable(err));
+      }
+
+      Logger.error(err);
+      callback();
+    }
+  }
+
+  _handleResult(result, callback) {
+    const resultIsPromise = result && result instanceof Promise;
+
+    if (!resultIsPromise) {
+      return callback(null, result);
+    }
+
+    this._resultAwaited = true;
+
+    let timeoutCalled = false;
+    const timeoutId = setTimeout(
+      () => {
+        this._resultAwaited = false;
+        timeoutCalled = true;
+        callback();
+      },
+      this._config.timeout
+    );
+
+    try {
+      result
```

Comment:
use await here to avoid having the .catch blocks

Reply:
Modified code to move from `then/catch` to `async/await`.